bundles/org.openhab.voice.actiontemplatehli/README.md

```diff
@@ -0,0 +1,389 @@
+
+You have an action with the template "play $* on living room" and another action with the template "play $musicAuthor on living room" and assuming 'mozart' is a valid value for the placeholder '$musicAuthor'.
+
+The sentence "play mozart on living room" will score 4 when compared with the template containing the dynamic placeholder and 5 when compared with the one without it.
+The action with template "play $musicAuthor on living room" will be executed.
+
+The sentence "play beethoven on living room" will score 4 when compared with the template containing the dynamic placeholder and 0 when compared with the one without it.
+The action with template "play $* on living room" will be executed.
+
+### POS Transformation
+
+POS is a technique which produces tags for each token, here though we are going to use it to match a group of words with a value so we should transform those words to a single token.
+That's the reason why the whitespace character should be replaced by '__' in the POS dictionaries and static values, you can see some examples bellow.
+
+### Target members:
+
+When the target of an action is a group item, you can target its members instead.
+
+You can use the following fields:
+
+* itemName: name of the item member to target. If present the other fields are ignored.
+* itemType: type of the item members to target.
+* requiredTags: allow to restrict the members targeted by tags when matching by type.
+* recursive: when matching by itemType, look for group members in a recursive way, default true.
+* mergeState: on a read action when matching by itemType, merge the item states by performing an AND operation, only allowed for 'Switch' and 'Contact' item types, default false.
+
+## Text Preprocessing
+
+The interpreter needs to match the input text with a target item and action configuration, to know what to do.
+To do so, it needs the tokens and optionally the POS tags and the lemmas.
+
+### Tokenizer
+
+You can provide a custom model at '<OPENHAB_USERDATA>/actiontemplatehli/token.bin', otherwise it will use the built-in simple tokenizer or whitespace tokenizer (configurable).
+
+Here you have an example of the built-in ones:
+
+* Using the white space tokenizer "What time is it?" produces the tokens "what" "time" "is" "it?"
+* Using the simple tokenizer "What time is it?" produces the tokens "what" "time" "is" "it" "?"
+
+Tokenizing the text is enough to use the action type 'tokens' as tokens are the only ones required for scoring (but the option 'optionalLanguageTags' will not take effect unless you have the POS language tags).
+
+### POSTagger (language tags)
+
+You need to provide a model for POS tagging at '<OPENHAB_USERDATA>/actiontemplatehli/pos.bin' for your language. 
+This will produce a language tag for each token, that can be used in 'optionalLanguageTags' to make some optional for scoring.
+As an example:
+
+The tokens "that,sounds,good" produces the tags "DT,VBZ,JJ".
```

Comment:
How can you guess the values of these tags ?

Reply:
Those depends on the lemmatization model you are using.

Reply:
I can find this link https://www.ling.upenn.edu/courses/Fall_2003/ling001/penn_treebank_pos.html 
But I think there is not an official standard for those, and that can different by language.

Reply:
I will add a note to recall this.